proto/bootz.proto

```diff
@@ -0,0 +1,167 @@
+
+import "github.com/openconfig/gnsi/authz/authz.proto";
+import "github.com/openconfig/gnsi/certz/certz.proto";
+import "github.com/openconfig/gnsi/credentialz/credentialz.proto";
+import "github.com/openconfig/gnsi/pathz/pathz.proto";
+
+
+option go_package = "github.com/openconfig/bootz/proto/bootz";
+
+
+// The Bootz service definition.
+// Bootz service is hosted by the bootz-server, and the device is the
+// client (as opposed to most g*i services, which are hosted by the device).
+service Bootstrap {
+  // This RPC returns the data required to put the device in a known state
+  // (i.e. OS, bootloader password, etc) and applies an initial device
+  // configuration.
+  rpc GetBootstrapData(GetBootstrapDataRequest) returns 
+    (GetBootstrapDataResponse) {}
+   
+  // This RPC is used by the device to indicate successful application of
+  // BootstrappingData. The Bootz system will proceed to the next step if
+  // a SUCCESS is reported, otherwise it will retry or put the device in an
+  // out-of-service state. The device should validate the server's identity
+  // against the server_trust_cert it obtained in GetBootstrappingDataResponse.
+  rpc ReportStatus(ReportStatusRequest) returns (EmptyResponse) {}
+}
+
+// Initial request from device to the bootz-server. Contains a detailed
+// description of the chassis and its constituent control cards.
+message GetBootstrapDataRequest {
+  ChassisDescriptor chassis_descriptor = 1;
+  ControlCardState control_card_state = 2;
+  // A nonce that the bootstrap server should use when signing the response.
+  string nonce = 1001;
+}
+
+// Structured message containing the chassis details, and listing its component
+// control-card modules.
+message ChassisDescriptor {
+  string manufacturer = 1;
+  string part_number = 2;
+  // This field is populated for fixed form-factor chassis, and empty for
+  // modular chassis.
+  string serial_number = 3;
+  // For modular chassis this field will be used to list the control cards
+  // to be bootstrapped.
+  repeated ControlCard control_cards = 4;
+}
+
+// Details of the control card, including serial-number and the location
+// of the card on the chassis.
+message ControlCard {
+  string part_number = 1;
+  string serial_number = 2;
+  int32 slot = 3;
+}
+
+// The status of each control card (i.e. initialized or not). For first install
+// of the chassis, both control cards are in NOT_INSTALLED state. For 
+// install a hot-swapped standby, the status of the active card is INSTALLED, 
+// and that of the standby is NOT_INSTALLED. For fixed form-factor chassis,
+// the serial number is that of the chassis, and status is NOT_INSTALLED.
+message ControlCardState {
+  string serial_number = 1;
+  ControlCardStatus status = 2;
+}
+
+enum ControlCardStatus {
+  CONTROL_CARD_STATUS_UNSPECIFIED = 0;
+  CONTROL_CARD_STATUS_NOT_INSTALLED = 1;
+  CONTROL_CARD_STATUS_INSTALLED = 2;
+}
+
+message GetBootstrapDataResponse {
+  message BootstrapDataResponse {
```

Comment:
Should we add the nonce in here? The nonce needs to be part of the signed message

Reply:
where do you think it shouldbe added?

Reply:
As last field within  message BootstrapDataResponse e.g.:
string nonce = 1001;

Reply:
But ideally, i would create a new message to avoid having it in the repeated filed:

message BootstrapDataResponseSigned {
     repeated BootstrapDataResponse responses = 1;
     string nonce = 1001;
}

In this case we solve 2 problems:
1. the nonce is set only once.
2. The signed message is clear and confined (Easily computed from bytes)